tests/test_generate_database_data.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+import mock
+from dmscripts.generate_database_data import generate_user, USER_ROLES
+
+TEST_ROLES = USER_ROLES + ["not-a-role"]
+
+
+class TestGenerateUser:
+
+    def setup_method(self, method):
+        self.api_client_patch = mock.patch("dmapiclient.DataAPIClient", autospec=True)
+        self.api_client = self.api_client_patch.start()
+
+    def teardown_method(self, method):
+        self.api_client_patch.stop()
+
+    def test_generate_user_has_correct_keys(self):
+        buyer = generate_user(data=self.api_client, role="buyer")
+        assert set(buyer.keys()) == {"name", "email_address", "password", "role"}
```

Comment:
Nit: I don't think the `set` is needed - the output of `keys` is already set-like:

```
In [6]: {1:2, 3:4}.keys() == {3, 1}                                                                                                                                                            
Out[6]: True
```